crates/typst/src/layout/grid/layout.rs

```diff
@@ -1083,31 +1243,19 @@ impl<'a> GridLayouter<'a> {
         self.measure_columns(engine)?;
 
         for y in 0..self.grid.rows.len() {
-            // Skip to next region if current one is full, but only for content
-            // rows, not for gutter rows, and only if we aren't laying out an
-            // unbreakable group of rows.
-            let is_content_row = !self.grid.is_gutter_track(y);
-            if self.unbreakable_rows_left == 0 && self.regions.is_full() && is_content_row
-            {
-                self.finish_region(engine)?;
-            }
-
-            if is_content_row {
-                // Gutter rows have no rowspans or possibly unbreakable cells.
-                self.check_for_rowspans(y);
-                self.check_for_unbreakable_rows(y, engine)?;
-            }
-
-            // Don't layout gutter rows at the top of a region.
-            if is_content_row || !self.lrows.is_empty() {
-                match self.grid.rows[y] {
-                    Sizing::Auto => self.layout_auto_row(engine, y)?,
-                    Sizing::Rel(v) => self.layout_relative_row(engine, v, y)?,
-                    Sizing::Fr(v) => self.lrows.push(Row::Fr(v, y)),
+            if let Some(header) = &self.grid.header {
+                if y < header.end {
+                    if y == 0 {
+                        // Use the appropriate function to layout header
+                        // rows.
```

Comment:
```suggestion
                        // Use the appropriate function to layout header rows.
```

Reply:
on second look, this comment doesn't add any information to the code, so it could just as well be removed I think